pkg/source/source.go

```diff
@@ -4,24 +4,32 @@ import (
 	"fmt"
 )
 
-const doxIdFmt = "dox: %s"
-const doxIgnore = "dox: ignore"
-const doxOmitNotice = "dox: omit-notice"
+const doxHeaderFmt = "dox: %s"
+const doxHeaderRegexp = `dox: (.*)`
+
+// source directives (SD)
+const (
```

Comment:
Yeah, I think so. Were you thinking we should have a `Header` struct with functions to serialize and deserialize?

Reply:
After more thought, the dox header is providing identification and directives how the the source should be published. Give me another day to think through how to best represent this.

Reply:
Sure. I have been playing around with this for a bit but haven't come up with anything good yet. There are some difficulties, like the page ID, which I don't think should be called a directive. Then, as you have brought up, we should have a solution that could handle directives with values other than bool. 

My initial attempt used a header struct, an omitNotice struct, and an ignore struct. Header struct had fields for omitNotice and ignore. Methods would have to handle the directive structs explicitly, so this solution isn't scalable. I'm not sure how an interface would work either, since we could be dealing with different value types (`directive.Value()` -> bool, `directive.Value()` -> string)